server/src/main/java/org/opensearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -2891,17 +2943,21 @@ private BlobPath shardPath(IndexId indexId, int shardId) {
     /**
      * Configures RateLimiter based on repository and global settings
      *
-     * @param repositorySettings repository settings
-     * @param setting            setting to use to configure rate limiter
-     * @param defaultRate        default limiting rate
+     * @param lastRateLimiter last rate limiter
+     * @param bytesPerSecSetting setting to use to configure rate limiter
+     * @param settings        repository settings
      * @return rate limiter or null of no throttling is needed
      */
-    private RateLimiter getRateLimiter(Settings repositorySettings, String setting, ByteSizeValue defaultRate) {
-        ByteSizeValue maxSnapshotBytesPerSec = repositorySettings.getAsBytesSize(setting, defaultRate);
-        if (maxSnapshotBytesPerSec.getBytes() <= 0) {
+    private RateLimiter getRateLimiter(RateLimiter lastRateLimiter, Setting<ByteSizeValue> bytesPerSecSetting, Settings settings) {
+        // if new value is not set, return last rate limiter
+        if (lastRateLimiter != null && bytesPerSecSetting.exists(settings) == false) {
+            return lastRateLimiter;
+        }
+        ByteSizeValue maxByteSize = bytesPerSecSetting.get(settings);
+        if (maxByteSize.getBytes() <= 0) {
```

Comment:
Shouldn't zero value mean rate limited to zero bytes instead of no rate limiter?

Reply:
I don't make any changes here, just keep the consistent with the previous, I confirm that when `indices.recovery.max_bytes_per_sec`=0, there is no no rate limiter too. So it seems that keeping the logic is feasible.